drivers/infiniband/core/uverbs_std_types.c

```diff
@@ -1116,6 +1116,104 @@ int uverbs_modify_qp_handler(struct ib_device *ib_dev,
 	return ret;
 }
 
+
+DECLARE_UVERBS_ATTR_SPEC(
+	uverbs_query_qp_spec, 
+	UVERBS_ATTR_IDR(QUERY_QP_HANDLE,
+			UVERBS_TYPE_QP,
+			UVERBS_ACCESS_READ,
+			UA_FLAGS(UVERBS_ATTR_SPEC_F_MANDATORY)),
+	UVERBS_ATTR_PTR_IN(QUERY_QP_ATTR_MASK, u32), 
```

Comment:
I guess it would be better to split the response to the actual fields we care of (instead of getting an attr_mask). In that way, you could just ask if a field exists, query and output this specific data.